molgenis-data/src/main/java/org/molgenis/util/EntityUtils.java

```diff
@@ -442,6 +442,7 @@ public static boolean equals(Entity entity, Entity otherEntity)
 							.equals(otherXrefValue.getIdValue())) return false;
 					break;
 				case CATEGORICAL_MREF:
+				case ONE_TO_MANY:
```

Comment:
Can you mention this bug in your pull request description? Or even better, perform a separate pull request for this bug fix? Or even better then better perform a separate pull request for this bug and add a unit test for one-to-many.